src/lib/helpers.ts

```diff
@@ -1,5 +1,24 @@
 import type { User } from "@/types/api";
+import type { SelectInputOption } from "@/types/forms";
 
 /** Prefers the user's full name, falling back to their email. */
 export const getUserDisplayName = (user: User): string =>
   user.fullName ?? user.email;
+
+export function enumToFormSelectOptions<
+  T extends Record<string, string | number>,
+>(
+  enumObject: T,
+  labels: Partial<Record<T[keyof T], string>>,
+): SelectInputOption[] {
+  return Object.entries(enumObject)
+    .filter(([key]) => Number.isNaN(Number(key)))
+    .map(([, value]) => ({
+      value,
+      label: labels[value as T[keyof T]] ?? String(value),
+    }));
+}
```

Comment:
a i tutaj z tymi typami miałem na myśli takie coś, myślę że to wygląda bardziej clean


```suggestion
export const enumToFormSelectOptions = <T extends string | number>(
  enumObject: Record<string, T>,
  labels: Record<T, string>,
): SelectInputOption[] =>
  Object.entries(enumObject)
    .filter(([key]) => Number.isNaN(Number(key)))
    .map(([, value]) => ({
      value,
      label: labels[value],
    }));
```
(ewentualnie z tym `partial` i sprawdzaniem czy klucz istnieje, nie wiem czy to celowo dodałeś na przyszłość czy niepotrzebne)

i też jak mowiłem żebyś zmienił na zwykłą funkcję to był to momentowy skill issue bo jednolinijkowa funkcja jak najbardziej powinna być po prostu arrow constem bez klamer i returna